decent_whisper/backends/mlx_whisper.py

```diff
@@ -0,0 +1,158 @@
+import importlib.util
+from queue import SimpleQueue
+import threading
+from time import sleep
+from typing import Iterable, List, Optional, Tuple
+import numpy as np
+from numpy.typing import NDArray
+from threading import Thread
+
+from decent_whisper import Word, TranscriptionInfo
+from decent_whisper.model import ALL_LANGUAGES, TURBO_LANGUAGES, ModelInfo, path_for_model
+from .mlx_whisper_transcribe import transcribe as transcribe_mlx
+
+
+def name():
+    return "mlx_whisper"
+
+
+def should_use():
+    return importlib.util.find_spec("mlx") is not None
+
+
+def transcribe(
+    audio: NDArray[np.float64] | str, model: ModelInfo, language: Optional[str] = None
+) -> Tuple[Iterable[List[Word]], TranscriptionInfo]:
+    queue = SimpleQueue()
+    error_event = threading.Event()
+    exception = None
+
+    def work():
+        def on_new_segment(segment: dict):
+            queue.put(segment)
+
+        def on_language_detected(lang: str):
+            nonlocal language
+            language = lang
+
+        try:
+            transcribe_mlx(
+                audio,
+                path_or_hf_repo=str(path_for_model(model)),
+                language=language,
+                word_timestamps=True,
+                on_new_segment=on_new_segment,
+                on_language_detected=on_language_detected,
+                silent=True,
+                hallucination_silence_threshold=0.5,
+            )
+            queue.put(None)
+        except Exception as e:
+            nonlocal exception
+            exception = e
+            error_event.set()
+            queue.put(None)
+
+    thread = Thread(target=work)
+    thread.start()
+
+    def map_segments():
+        while True:
+            segment = queue.get()
+            if segment is None:
+                if error_event.is_set():
+                    raise exception or RuntimeError("Transcription failed")
+                break
+
+            yield [
+                Word(
+                    start=word["start"],
+                    end=word["end"],
+                    probability=word["probability"],
+                    word=word["word"]
+                )
+                for word in segment["words"]
+            ]
+
+        thread.join()
+
+    while language is None:
+        if error_event.is_set():
+            raise exception or RuntimeError("Transcription failed")
+
+        sleep(0.1)
```

Comment:
I think this would break usage in the transcribee worker, because it blocks the GIL and we rely on the asyncio Thread Executor to be able to run the keepalive task.

Reply:
Why? Sleep releases the GIL, so it should be fine?

Reply:
Ah true, I thought it did not, because there is a `asyncio.sleep`